src/config.py

```diff
@@ -0,0 +1,348 @@
+        return self.variablesDef
+
+    @property
+    def routes(self):
+        """Return the routes definition.
+
+        :return: array
+            The routes array, in PHP nested array form.
+        :exception RuntimeException:
+            If the routes are not accessible due to being in the wrong environment.
+        """
+        if not self.is_valid_platform():
+            raise RuntimeError('You are not running on Platform.sh, so routes are not available.')
+        if self.in_build():
+            raise RuntimeError('Routes are not available during the build phase.')
+        return self.routesDef
+
+    def get_route(self, route_id):
+        """Get route definition by route ID.
+
+        :param route_id: string
+            The ID of the route to load.
+        :return: array
+            The route definition. The generated URL of the route is added as a 'url' key.
+        :exception ValueError:
+            If there is no route by that ID, an exception is thrown.
+
+        """
+
+        for (url, route) in self.routes.items():
+            if route['id'] == route_id:
+                route['url'] = url
+                return route
+        raise ValueError('No such route id found: {}'.format(route_id))
+
+    def application(self):
+        """Returns the application definition array.
+
+        This is, approximately, the .platform.app.yaml file as a nested array. However, it also has other information
+        added by Platform.sh as part of the build and deploy process.
+
+        :return: array
+            The application definition array.
+        """
+
+        if not self.is_valid_platform():
+            raise RuntimeError('You are not running on Platform.sh, so the application definition are not available.')
+        return self.applicationDef
+
+    def on_enterprise(self):
+        """Determines if the current environment is a Platform.sh Enterprise environment.
+
+        :return: bool
+            True on an Enterprise environment, False otherwise.
+        """
+
+        return self.is_valid_platform() and self._get_value('MODE') == 'enterprise'
+
+    def on_production(self):
+        """Determines if the current environment is a production environment.
+
+        Note: There may be a few edge cases where this is not entirely correct on Enterprise, if the production
+        branch is not named `production`. In that case you'll need to use your own logic.
+
+        :return: bool
+            True if the environment is a production environment, False otherwise.
+            It will also return False if not running on Platform.sh or in the build phase.
+
+        """
+
+        if not self.is_valid_platform() and not self.in_build():
+            return False
+        prod_branch = 'production' if self.on_enterprise() else 'master'
+        return self._get_value('BRANCH') == prod_branch
+
+    def __get_value(self, name):
```

Comment:
On second thought, since this is a purely internal method, I'd probably just implement `__getitem__` instead. Then, instead of calling `self.__get_value("APPLICATION")`, you can just say `self["APPLICATION"]`.

Any objections to doing a purely internal method in a more Pythonic way, @Crell?

In any case, it looks like you're using a double underscore for the method name, but a single underscore everywhere you're calling it. Fix the name here

Reply:
Note that implementing this would also mean that users of the package could say `config["WHATEVER"]` when using this module in their code

Reply:
Purely internal code should definitely be done in the way that will impress users of that language the most, whatever that is. :smile: 

My only concerns would be "leaking" data (which since this is Python happens anyway by design) or possible confusion/collssion with `getattr`.  As long we're safe/comfortable on that front I've no objection.

Reply:
`__getattr__` is for dots, `__getitem__` is for brackets. They could theoretically return totally different things, like I like `config.project` to get the user-facing project attribute, with `self["WHATEVER"]` for the internals. Users could technically do `self["WHATEVER"]` if they wanted to as well, but that's on them for choosing to use a well-documented lib incorrectly